apps/web/vibes/soul/sections/product-detail/index.tsx

```diff
@@ -99,15 +99,23 @@ export function ProductDetail<F extends Field>({
 
                   <Stream fallback={<ProductDetailFormSkeleton />} value={streamableFields}>
                     {(fields) => (
-                      <ProductDetailForm
-                        action={action}
-                        ctaLabel={ctaLabel}
-                        decrementLabel={decrementLabel}
-                        fields={fields}
-                        incrementLabel={incrementLabel}
-                        productId={product.id}
-                        quantityLabel={quantityLabel}
-                      />
+                      <Stream
+                        fallback={<ProductDetailFormSkeleton />}
+                        value={streamableSubmitButton}
+                      >
```

Comment:
This feels off to me. We've got way to many uses of `<Stream>` and granular skeletons. We should combine streamable with `Promise.all` and use a single `<Stream>`

Reply:
I agree, will try to change this.

Reply:
Fixed.